docs/clients/promtail/stages/metrics.md

```diff
@@ -70,6 +70,13 @@ type: Gauge
 # defaulting to the metric's name if not present.
 [source: <string>]
 
+# Label values on metrics are dynamic which can cause exported metrics
+# to go stale (for example when a stream stops receiving logs).
+# To prevent unbounded growth of the /metrics endpoint any metrics which
+# have not been updated within this time will be removed.
+# Must be greater than or equal to '1s', if undefined default is '5m'
+[max_idle_duration: <string>]
```

Comment:
would it be easier to have a global config instead of a per metric ?

Reply:
Not from an implementation perspective, we have no precedent for passing promtail configs into a pipeline.  

I also think it makes more sense here, you can define as few or as many metrics as you would like in a metrics stage and being able to change idle timeout for a specific metric may be advantageous for someone?